packages/internal/src/__tests__/validateSchema.test.ts

```diff
@@ -7,22 +7,78 @@ const FIXTURE_PATH = path.resolve(
   '../../../../__fixtures__/example-todo-main'
 )
 
-beforeAll(() => {
-  process.env.RWJS_CWD = FIXTURE_PATH
+const FIXTURE_ERROR_PATH = path.resolve(
```

Comment:
Would it not be possible to use the same fixture here instead of creating a new one for this small error?

Reply:
@dac09 I'm not sure I can. The validation is sort of all or nothing -- it completely succeeds or fails.

There's no way (I don't think) to know why it failed.

```ts
    try {
      await loadAndValidateSdls()
      isSdlValid = true
    } catch (e) {
      isSdlValid = false
    }
```

Thus, I need SDL's that test 


```
   createTodo(body: String!): Todo @requireAuth(roles: 12)
    updateTodoStatus(id: Int!, status: String!): Todo @requireAuth(roles: ["admin", 12])
    renameTodo(id: Int!, body: String!): Todo @requireAuth(roles: [null, 12])
```

I know it seems like a lot for just the SDL. 

I can remove the web side though to reduce the number of files.

Reply:
Is it not possible to have this in the example_todo_with_errors fixture? It should just be one (or a few) extra lines in one of the SDLS there for you to be able to test

Reply:
So, my understanding and I could be wrong is that `loadAndValidateSdls` loads *all* SDLs in the api.

Therefore, I cannot have some flavors of the requireAuth in one SDL and something else -- it's all or nothing.

Or I guess I could use this 

```ts
export const loadAndValidateSdls = async () => {
  const projectTypeSrc = await loadTypedefs(
    ['graphql/**/*.sdl.{js,ts}', 'directives/**/*.{js,ts}'],
    {
      loaders: [
        new CodeFileLoader({
          noRequire: true,
          pluckConfig: {
            globalGqlIdentifierName: 'gql',
          },
        }),
      ],
      cwd: getPaths().api.src,
    }
  )

  // The output of the above function doesn't give us the documents directly
  const projectDocumentNodes = Object.values(projectTypeSrc)
    .map(({ document }) => document)
    .filter((documentNode): documentNode is DocumentNode => {
      return !!documentNode
    })

  // Merge in the rootSchema with JSON scalars, etc.
  const mergedDocumentNode = mergeTypeDefs([
    rootSchema.schema,
    projectDocumentNodes,
  ])

  validateSchemaForDirectives(mergedDocumentNode)
}
```

as a guide and load individual slides and make my own .... and really just test `validateSchemaForDirectives` for one SDL file

Reply:
Ok, I reworked the tests to work on individual SDLs so we can reuse one test project with errors

Reply:
@dac09 ^^^

Reply:
Thanks a bunch DT! Looking much nicer now